src/data/repositories/DataSetD2Api.ts

```
@@ -208,7 +210,20 @@ export class DataSetD2Api {
         const projectDetails = projects.find(project => project.id === projectAttributeId);
         const dataElementGroups = this.buildDataElementsGroupsCodes(d2DataSet);
 
+        const disabledFields = d2DataSet.sections.flatMap(section => {
+            const degCode = this.extractCompetencyCode(section.id, section.code);
+            const coreCompetency = coreCompetencies.find(cc => cc.code === degCode);
+            return section.greyedFields.map((greyField): DataSet["disabledFields"][number] => {
+                return {
+                    competencyId: coreCompetency?.id ?? "",
```

Probably, that's not a problem here, but falling back to "" should always be considered carefully. There are different scenarios. In the scenario "this is something that should never happen," it's better to throw an exception (it's safer and keeps the code cleaner).